src/main/java/yoyo/DialogBox.java

```diff
@@ -0,0 +1,65 @@
+package yoyo;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+/**
+ * Represents a dialog box consisting of an ImageView to represent the speaker's
+ * face and a label containing text from the speaker.
+ */
+public class DialogBox extends HBox {
+
+    @FXML
+    private Label dialog;
+    @FXML
+    private ImageView displayPicture;
+
+    private DialogBox(String text, Image img) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBox.fxml"));
+            fxmlLoader.setController(this);
+            fxmlLoader.setRoot(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        dialog.setText(text);
+        displayPicture.setImage(img);
+    }
```

Comment:
I like how you've implemented the FXML loading pattern within the component's own constructor. This self-contained approach is clean and makes the class easy to instantiate. he constructor currently handles the IOException by printing the stack trace. While this works, it might leave the UI in a partially initialized state if the FXML file is missing. Have we considered making the exception more explicit, for example by re-throwing it as an UncheckedIOException or a custom InitializationException? This would force the caller to handle this critical error, making the application more robust.
```suggestion
    private DialogBox(String text, Image img) {
    try {
        FXMLLoader fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBox.fxml"));
        fxmlLoader.setController(this);
        fxmlLoader.setRoot(this);
        fxmlLoader.load();
    } catch (IOException e) {
        // Wrap and re-throw to avoid a partially constructed object
        throw new RuntimeException("Failed to load FXML for DialogBox", e);
    }
    // ... rest of initialization
}